assets/javascripts/modules/renew/renew.es6

```diff
@@ -81,20 +84,39 @@ export function send(state, errorHandler) {
         if(promoCode){
             payload.promoCode = promoCode;
         }
-            ajax({
-            type: 'json',
+        let request = new Request('/manage/renew', { 
             method: 'POST',
-            url: '/manage/renew',
-            headers: {
-                'Csrf-Token': document.querySelector('input[name="csrfToken"]').getAttribute('value')
-            },
-            contentType: 'application/json',
-            data: JSON.stringify(payload)
-        }).then((response) => {
-            window.location.assign(response.redirect);
+            body: JSON.stringify(payload),
+            headers: new Headers({
+                'Csrf-Token': document.querySelector('input[name="csrfToken"]').getAttribute('value'),
+                'Content-Type':  'application/json'
+            }),
+            credentials: 'include'
+        })
+        fetch(request).then((response) => {
+            if (response.status == 200) {
+                console.log(response)
+                response.json().then((json) => {
+                    window.location.assign(json.redirect);
+                }).catch((e) => {
+                    Raven.captureException(e)
+                    //We got a 200, and then couldn't parse the JSON.
+                    //The renewal probably worked, and we should not really be in this branch.
+                    errorHandler(renewNetworkErrorMessage)
+
+                })
+                return;
+            }
+            Raven.captureException(`Renewal failed with status code ${response.status}`);
+            if (response.status == 500 || response.status == 400) {
+                errorHandler(renewErrorMessage);
+                return;
+            }
+            errorHandler(renewNetworkErrorMessage)
+            return;
         }).catch((r) => {
```

Comment:
So this case will capture network timeouts etc.? Out of interest, what timeout do we use for this request?

Reply:
It's quite possible that the cases we were seeing where users saw the failure screen but the renew actually succeeded were due to us giving up on waiting for a response, whilst members-data-api was still processing the request.

Reply:
I think it's likely that is the case. 

I think the timeout we're likely to be hitting is the fastly one. IIRC we increased it to match the Zuora one ages ago, but this could be an expensive enough call to be a problem.